### PR TITLE
Add mina precompiles

### PR DESCRIPTION
**Pull request by MartinOndejka**

### CODE REVIEW

File: core/vm/mina_contracts.go [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+
+var sol_bool, _ = abi.NewType("bool", "", nil)
+var sol_uint8, _ = abi.NewType("uint8", "", nil)
+var sol_bytes32, _ = abi.NewType("bytes32", "", nil)
+var sol_bytes32Arr, _ = abi.NewType("bytes32[]", "", nil)
+
+type MinaPoseidon struct{}
+
+func (c *MinaPoseidon) RequiredGas(input []byte) uint64 {
+	return 1000
```
Comment (SebastienGllmt) [inline core/vm/mina_contracts.go]: Was there any rationale for this number? I don't think it really matters that much, but just to know

File: core/vm/mina_contracts.go [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+		return nil, ErrExecutionReverted
+	}
+
+	return output_buffer[:], nil
+}
+
+type MinaSigner struct{}
+
+func (c *MinaSigner) RequiredGas(input []byte) uint64 {
+	return 1000
```
Comment (SebastienGllmt) [inline core/vm/mina_contracts.go:119]: Same thing here

File: core/vm/mina_contracts.go (hunk 1 above)
Comment (MartinOndejka) [reply core/vm/mina_contracts.go]: No, just placeholder for now

File: core/vm/mina_contracts.go (hunk 2 above)
Comment (MartinOndejka) [reply core/vm/mina_contracts.go:119]: Same, just placeholder for now.

**Review by SebastienGllmt — APPROVED** [review]

**Pull request merged**